src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -203,29 +204,52 @@ private function createRelationsForItem($item, $formattedData)
      */
     private function getRelationDataFromFormData($data)
     {
-        $relationFields = $this->getRelationFields();
+        $relation_fields = $this->getRelationFields();
 
         $relationData = [];
-        foreach ($relationFields as $relationField) {
-            $attributeKey = $relationField['name'];
-            if (array_key_exists($attributeKey, $data) && empty($relationField['pivot'])) {
-                $key = implode('.relations.', explode('.', $relationField['entity']));
+        foreach ($relation_fields as $relation_field) {
+            $relation_field['name'] = $relation_field['entity'];
+            $attributeKey = $relation_field['name'];
+
+            if (! is_null(Arr::get($data, $attributeKey)) && empty($relation_field['pivot'])) {
+                $key = implode('.relations.', explode('.', $this->getOnlyRelationEntity($relation_field)));
                 $fieldData = Arr::get($relationData, 'relations.'.$key, []);
 
                 if (! array_key_exists('model', $fieldData)) {
-                    $fieldData['model'] = $relationField['model'];
+                    $fieldData['model'] = $relation_field['model'];
                 }
 
                 if (! array_key_exists('parent', $fieldData)) {
-                    $fieldData['parent'] = $this->getRelationModel($relationField['entity'], -1);
+                    $fieldData['parent'] = $this->getRelationModel($relation_field['entity'], -1);
                 }
-
-                $fieldData['values'][$attributeKey] = $data[$attributeKey];
+                $relatedAttribute = Arr::last(explode('.', $attributeKey));
+                $fieldData['values'][$relatedAttribute] = Arr::get($data, $attributeKey);
 
                 Arr::set($relationData, 'relations.'.$key, $fieldData);
             }
         }
 
         return $relationData;
     }
+
```

Comment:
What does this do again @pxpm ?

Reply:
Check if now you can understand from the description. This creates and array of relatations with nested relations if they happen to be.

For example, you have and Model with HasOne Address (line_1, country_id) and Address has a relation BelongsTo with Country.

This will create a relation array like this:

```php
  "relations" => array:2 [▼
    "addressless" => array:4 [▼
      "model" => "App\Models\Address"
      "parent" => "App\Models\Address"
      "values" => array:2 [▼
        "name" => "Av. Aure"
        "line_1" => "Rua Soei"
      ]
      "relations" => array:1 [▼
        "country" => array:3 [▼
          "model" => "App\Models\Country"
          "parent" => "App\Models\Address"
          "values" => array:1 [▼
            "country" => "1"
          ]
        ]
      ]
    ]
```

Later we iterate through this relations and create them.